### DISCUSSION TITLE
Can some one give me an idea regarding the USB packet data sequence?

### DISCUSSION THREAD

**Discussion by JordanChiang** [libusb/hidapi#566] [Q&A]
For example "HidD_GetManufacturerString", how to achieve by send USB_Setup_Packet to HID then read data back? 
I trace the USB signal by "Bus Hound", and the packet data like below, 
Type:0x80
Request:0x06
Value:0x0302
Index:0x0409
Length:0x28
But no idea which function should call, please advice, many thanks.

**Comment by JoergAtGithub** [comment]
Just call https://libusb.info/hidapi/group__API.html#ga702f45b42d85599eedd602c05661fde5 this returns the Manufacturer String from the USB String Descriptor.
The HID protocol itself can not return strings, it only provides the index of a string in the USB string register.

**Comment by JordanChiang** [comment]
Hi @JoergAtGithub ,
Thanks for the feedback, I know the MFG string have exist method to read, but I have customize command/data need to read/write by send USB setup packet, so any suggestion? thanks.
